### ISSUE TITLE
Functions longer than 512 tokens

### ISSUE THREAD

**Issue by cluosh** [CUMLSec/trex#10]
Hi,

for pretraining, having functions larger than 512 tokens seems trivial as they can just be split.

However, for similarity the paper states `We average the subsequences’ embeddings during
finetuning if the function is split to more than one subsequences.` How exactly does this work? I could not find the code for the averaging, and all input data appears to be < 512 tokens, which makes it seem like the functions have been split before; if the functions have been split before finetuning, how are the pairs matched in the dataset?

**Comment by peikexin9** [comment]
Hi @cluosh, thanks for noting that. We decided not to include that in our code release as that unnecessarily makes the code hard to read and parse, and we do not observe obvious performance change with the current strategy. So yes, for now we already split the functions and treat the pair as positive as long as they are from the same function, no matter which split it comes from.

**Comment by cluosh** [comment]
Some functions are gonna have a different number of splits, because an obfuscated version would usually be longer; are you handling that difference in a special way, or are you only taking the first n splits?

**Comment by peikexin9** [comment]
what I did is like I split any function longer into blocks with <512 tokens. For example, say func1 and func2 are similar (compiled from same function). Then func1: block1, block2, func2: block1, block2. Then any pair from these two functions are treated as positive samples, e.g., <func1_block1, func2_block2>, <func1_block2, func2_block2>, <func1_block1, func2_block1>, <func1_block2, func2_block1> all have positive labels.

**Comment by cluosh** [comment]
Okay, that's kind of what i had in mind, thanks for the replies!
